internal/pkg/manifest/backend_svc.go

```diff
@@ -114,16 +114,22 @@ func (s *BackendService) Publish() []Topic {
 	return s.BackendServiceConfig.PublishConfig.publishedTopics()
 }
 
-// BuildRequired returns if the service requires building from the local Dockerfile.
-func (s *BackendService) BuildRequired() (bool, error) {
-	return requiresBuild(s.ImageConfig.Image)
-}
-
 // BuildArgs returns a docker.BuildArguments object for the service given a context directory.
-func (s *BackendService) BuildArgs(contextDir string) map[string]*DockerBuildArgs {
+func (s *BackendService) BuildArgs(contextDir string) (map[string]*DockerBuildArgs, error) {
+	required, err := requiresBuild(s.ImageConfig.Image)
+	if err != nil {
+		return nil, err
+	}
 	buildArgs := make(map[string]*DockerBuildArgs, len(s.Sidecars)+1)
-	buildArgs[aws.StringValue(s.Name)] = s.ImageConfig.Image.BuildConfig(contextDir)
-	return buildArgs
+	if required {
+		buildArgs[aws.StringValue(s.Name)] = s.ImageConfig.Image.BuildConfig(contextDir)
+	}
+	for name, config := range s.Sidecars {
```

Comment:
I'm sure it is difficult to write only one function to do what we are doing in this `BuildArgs` throughout all our service types. I think we can separate out this side loop and put it in one function so that we reduce duplication. Let me know what do you think about it.

Reply:
created a function `buildArgs` to reduce code duplication.